src/app/store.service.ts

```diff
@@ -18,9 +28,82 @@ export class StoreService {
   constructor(
     private packageService: PackageService,
     private itemService: ItemService,
-    private authService: AuthService
+    private authService: AuthService,
+    private http: HttpClient
   ) { }
 
+  getAddressFromBackend(backendAddress: string): Address {
+    const splitAddr = backendAddress.split('\n')
+      return {
+        addressLine1: splitAddr[0],
+        addressLine2: splitAddr[1],
+        postalCode: splitAddr[2]
+    }
+  }
+
+  mockLogistics(): Logistics {
+    return {
+      partner: {
+        id: 'ninjavan'
+      },
+      pickUpAddress: {
+        addressLine1: '',
+        addressLine2: '',
+        postalCode: '',
+        useBusinessAddress: true
+      }
+    }
+  }
+
+  mockBusiness(): Business {
+    return {
+      name: 'Not Starbucks',
+      description: 'This store is not Starbucks.',
+      type: 'Food and Beverage'
+    }
+  }
+
+  async mapPackageItemToStorePackageItem(packageItem: PackageItem, merchantId: string): Promise<StorePackageItem> {
+    return {
+      item: await this.itemService.getItem(merchantId, packageItem.itemId),
+      quantity: packageItem.quantity
+    }
+  }
+
+  async mapPackageToStorePackage(inputPackage: Package, backEndStore: BackEndStore): Promise<StorePackage> {
+    return {
+      id: inputPackage.id,
+      merchantId: inputPackage.merchantId,
+      merchantName: backEndStore !== null ? backEndStore.name : this.mockBusiness.name,
+      name: inputPackage.name,
+      description: inputPackage.description,
+      cycle: inputPackage.cycle,
+      imageUrl: inputPackage.imageUrl,
+      rating: {
+        score: Math.floor((Math.random() * 5) + 3),
+        count: Math.floor((Math.random() * 100) + 1)
+      },
+      subscription: inputPackage.subscriptionPlans && inputPackage.subscriptionPlans.length > 1 ? {
```

Comment:
We'll need this object regardless of the number of subscription plans available.

If subscription plans is empty or undefined:
{
    basicPlanId: undefined,
    mostPopularId: undefined,
    plans: []
}